src/theme/components/featuredContent.ts

```diff
@@ -1,75 +1,84 @@
+import { createMultiStyleConfigHelpers } from "@chakra-ui/styled-system";
+import { StyleFunctionProps } from "@chakra-ui/system";
 import { wrapperStyles } from "./global";
 import { screenreaderOnly } from "./globalMixins";
 
-interface FeaturedContentBaseStyleProps {
+// This function creates a set of function that helps us
+// create multipart component styles.
+const { defineMultiStyleConfig, definePartsStyle } =
+  createMultiStyleConfigHelpers(["imgWrapper", "text", "wrapper"]);
+
+interface FeaturedContentBaseStyleProps extends StyleFunctionProps {
   imagePosition: string;
   imageWidth: string;
   isFullWidth: boolean;
 }
 
-const FeaturedContent = {
-  baseStyle: ({
-    imagePosition,
-    imageWidth,
-    isFullWidth,
-  }: FeaturedContentBaseStyleProps) => {
-    let imageAtEnd = imagePosition === "end";
-    let full = isFullWidth === true;
-    let wrapperWidth = "50%";
-    if (imageWidth === "oneQuarter") {
-      wrapperWidth = "25%";
-    } else if (imageWidth === "oneThird") {
-      wrapperWidth = "33%";
-    } else if (imageWidth === "twoThirds") {
-      wrapperWidth = "66%";
-    } else if (imageWidth === "threeQuarters") {
-      wrapperWidth = "75%";
+const FeaturedContent = defineMultiStyleConfig({
+  baseStyle: definePartsStyle(
+    ({
+      imagePosition,
+      imageWidth,
+      isFullWidth,
+    }: FeaturedContentBaseStyleProps) => {
+      let imageAtEnd = imagePosition === "end";
+      let full = isFullWidth === true;
+      let wrapperWidth = "50%";
+      if (imageWidth === "oneQuarter") {
+        wrapperWidth = "25%";
+      } else if (imageWidth === "oneThird") {
+        wrapperWidth = "33%";
+      } else if (imageWidth === "twoThirds") {
+        wrapperWidth = "66%";
+      } else if (imageWidth === "threeQuarters") {
+        wrapperWidth = "75%";
+      }
+      return {
+        bgColor: "ui.bg.default",
+        left: full ? "50%" : "auto",
+        marginLeft: full ? "-50vw" : "auto",
+        marginRight: full ? "-50vw" : "auto",
+        position: "relative",
+        right: full ? "50%" : "auto",
+        width: full ? "100vw" : "100%",
+        _dark: {
+          bgColor: "dark.ui.bg.default",
+        },
```

Comment:
All of these should be in a `base` property "part" as well, and in `FeaturedContent`, it should be used as `styles.base`.